vlog4j-client/src/main/java/org/vlog4j/client/picocli/VLog4jClientMaterialize.java

```diff
@@ -0,0 +1,187 @@
+	private List<String> ruleFiles = new ArrayList<>();
+
+//  TODO
+//	Support graal rule files
+//	@Option(names = "--graal-rule-file", description = "Rule file(s) in graal syntax", required = true)
+//	private List<String> graalRuleFiles = new ArrayList<>();
+
+	@Option(names = "--log-level", description = "Log level of VLog (c++ library). One of: DEBUG, INFO, WARNING (default), ERROR.", required = false)
+	private LogLevel logLevel = LogLevel.WARNING;
+
+	@Option(names = "--log-file", description = "Log file of VLog (c++ library). VLog will log to the default system output by default", required = false)
+	private String logFile;
+
+	@Option(names = "--chase-algorithm", description = "Chase algorithm. RESTRICTED_CHASE (default) or SKOLEM_CHASE.", required = false)
+	private Algorithm chaseAlgorithm = Algorithm.RESTRICTED_CHASE;
+
+	@Option(names = "--timeout", description = "Timeout in seconds. Infinite by default", required = false)
+	private int timeout = 0;
+
+	@Option(names = "--query", description = "Positive not-ground Literals to query after materialization in rls syntax. Vlog4jClient will print the size of its extension", required = true)
+	private List<String> queryStrings = new ArrayList<>();
+
+	@ArgGroup(exclusive = false)
+	private PrintQueryResults printQueryResults = new PrintQueryResults();
+
+	@ArgGroup(exclusive = false)
+	private SaveQueryResults saveQueryResults = new SaveQueryResults();
+
+	// TODO save model
+	// @ArgGroup(exclusive = false)
+	// private SaveModel saveModel = new SaveModel();
+
+	@Override
+	public void run() {
+		ExamplesUtils.configureLogging();
+
+		if (!printQueryResults.isConfigOk()) {
+			printQueryResults.printErrorAndExit();
+		}
+
+		if (!saveQueryResults.isConfigOk()) {
+			saveQueryResults.printErrorAndExit();
+		}
+
+		// TODO
+		// if (!saveModel.isConfigOk()) {
+		// saveModel.printErrorAndExit();
+		// }
+
+		System.out.println("Configuration:");
+
+		/* Configure rules */
+		final KnowledgeBase kb = new KnowledgeBase();
+		for (String ruleFile : ruleFiles) {
+			try {
+				RuleParser.parseInto(kb, new FileInputStream(ruleFile));
+				System.out.println("  --rule-file: " + ruleFile);
+			} catch (FileNotFoundException e) {
+				System.out.println("File not found: " + ruleFile + ". " + e.getMessage());
+				System.exit(1);
+			} catch (ParsingException e) {
+				System.out.println("Failed to parse rule file: " + ruleFile + ". " + e.getMessage());
+				System.exit(1);
+			}
+		}
+
+		/* Configure queries (We throw parsing query errors before materialization) */
+		List<PositiveLiteral> queries = new ArrayList<>();
+		for (String queryString : queryStrings) {
+			try {
+				queries.add(RuleParser.parsePositiveLiteral(queryString));
+				System.out.println("  --query: " + queries.get(queries.size() - 1));
+			} catch (ParsingException e) {
+				System.out.println("Failed to parse query: " + queryString + ". " + e.getMessage());
+				System.exit(1);
```

Comment:
You could not exit, but print a message and continue with the other queries.

Reply:
Thank you